### ISSUE TITLE
New unit tests are noisy

### ISSUE THREAD

**Issue by dschlaep** [DrylandEcology/SOILWAT2#176]
Split from #174

**Comment by dschlaep** [comment]
For instance, on appveyor for PR #175 
```
./sw_test
[==========] Running 21 tests from 8 test cases.
[----------] Global test environment set-up.
...
[ RUN      ] SWFlowTempTest.SoilTemperatureInit
...
SOIL_TEMP FUNCTION ERROR: soil temperature max depth (70.00 cm) must be more than soil layer depth (295.00 cm)... soil temperature will NOT be calculated
...
[ RUN      ] SWFlowTempTest.MainSoilTemperatureFunction
 Depth of Layers 20.000000, MaxDepth 1.000000
...
SOIL_TEMP FUNCTION ERROR: soil temperature max depth ( 1.00 cm) must be more than soil layer depth (20.00 cm)... soil temperature will NOT be calculated
...
```

**Comment by dschlaep** [comment]
For instance, on travis-ci for PR #175:
```
[ RUN      ] SWFlowTempTest.SoilTemperatureInit
SOIL_TEMP FUNCTION ERROR: soil temperature max depth (70.00 cm) must be more than soil layer depth (295.00 cm)... soil temperature will NOT be calculated
[       OK ] SWFlowTempTest.SoilTemperatureInit (9 ms)
```

**Comment by CaitlinA** [comment]
The message 

> SOIL_TEMP FUNCTION ERROR: soil temperature max depth (70.00 cm) must be more than soil layer depth (295.00 cm)... soil temperature will NOT be calculated

is an error message generated from SOILWAT2 (Line 1571, SW_Flow_Lib.c), not the unit test. The unit test is an "EXPECT_DEATH" test and I expect that message to appear. 

Would you recommend I not do this test or do it in a different way? 

As an aside, appveyor, this message pops up:

> Death tests use fork(), which is unsafe particularly in a threaded context. For this test, Google Test couldn't detect the number of threads.


so maybe we should be avoiding or skipping death tests on appveyor anyway. What do you think?

**Comment by CaitlinA** [comment]
I will omit printf statements that generate other comments

**Comment by dschlaep** [comment]
> Re: I will omit printf statements that generate other comments

Thanks! And please always use `swprintf` in `SOILWAT` code (defined in `generic.h` for cross-compatibility between "pure" c/c++ and R-typed c)

**Comment by CaitlinA** [comment]
Copy. Also did you see the comment above about EXPECT_DEATH?

**Comment by dschlaep** [comment]
> Re: error message generated from SOILWAT2

I am wondering whether we would like to set `logfp` to `/dev/null` or similar during unit testing to silence `SOILWAT2`-generated messages during our unit tests?

> Re: appveyor message regarding `fork()`

Windows does multi-threading in its own way, i.e., it doesn't use `pthreads`. GoogleTest attempts to determine whether it runs with multiple threads (we are not), but it cannot affirm that on Windows. Hence it issues a warning at `fork()` which is used by `EXPECT_DEATH` -- but we can ignore that warning because we know that we run single threated (see https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#death-tests-and-threads)

**Comment by dschlaep** [comment]
Good point! Actually, I don't see the point of printing the same text via `swprintf` and via `LogError`.

I agree that wrapping the `swprintf` statement in an `#ifdef SWDEBUG` serves this issue; but maybe we should remove it altogether. Do you see any purpose of having it twice?
